examples/rn-mbnb/app/AirbnbList.tsx

```diff
@@ -0,0 +1,300 @@
+import { useApp, useUser } from "@realm/react";
+import React, { useCallback, useEffect, useState } from "react";
+import {
+  FlatList,
+  Text,
+  TextInput,
+  View,
+  StyleSheet,
+  Button,
+  ListRenderItem,
+  Pressable,
+  Platform,
+} from "react-native";
+import { useLocalQuery, useLocalRealm } from "./localRealm";
+import { SearchCache } from "./localModels";
+import { useSyncedQuery, useSyncedRealm } from "./syncedRealm";
+import { ListingsAndReview } from "./syncedModels";
+import FastImage from "react-native-fast-image";
+import RNFS from "react-native-fs";
+import { DefaultUserProfileData } from "realm";
+
+enum ResultMethod {
+  None = "Search Not Performed",
+  Cache = "Cache Hit",
+  Local = "Local Full Text Search",
+  Remote = "Atlas Search",
+}
+
+type SearchFunction = ({searchPhrase, pageNumber, pageSize}: {searchPhrase: string, pageNumber: number, pageSize: number}) => Promise<{error: string, result: ListingsAndReview[]}>;
+
+export const AirbnbList = () => {
+  // The results of the most recent search
+  const [resultIds, setResultIds] = useState([]);
+
+  // Offline mode is toggled in state and will effect how search is performed
+  const [offlineMode, setOfflineMode] = useState(false);
+
+  // The search term is stored in state
+  const [searchTerm, setSearchTerm] = useState("");
+
+  // Ids of all search results that are stored in Realm
+  const [cachedIds, setCachedIds] = useState([]);
+
+  // The method used to get the search results
+  const [resultMethod, setResultMethod] = useState<ResultMethod>(
+    ResultMethod.None
+  );
+
+  // Size of the synced and local Realm databases
+  const [syncedDbSize, setSyncedDbSize] = useState(0);
+  const [localDbSize, setLocalDbSize] = useState(0);
+
+  // Size of the image cache from react-native-fast-image
+  const [cacheSize, setCacheSize] = useState(0);
+
+  // Get the local Realm database for cache entries
+  const localRealm = useLocalRealm();
+
+  // Get the synced Realm database containing all subscribed listing items
+  const syncedRealm = useSyncedRealm();
+
+  // Get the current user in order to call the search function
+  const user = useUser<{searchListings: SearchFunction}, Record<string, unknown>, DefaultUserProfileData  >();
+
+  // Check the local cache for search results
+  const cache = useLocalQuery(
+    SearchCache,
+    (col) => col.filtered("searchTerm == $0", searchTerm.toLowerCase()),
+    [searchTerm]
+  );
+
+  // Get all cached search results in order to build our flexible sync subscription
+  const fullCache = useLocalQuery(SearchCache);
+
+  // Get the listings that match the search results ids
+  const listings = useSyncedQuery(
+    ListingsAndReview,
+    (col) => col.filtered("_id in $0", resultIds),
+    [resultIds]
+  );
```

Comment:
Same as previous comment.
```suggestion
  const listings = useSyncedQuery({
    type: ListingsAndReview,
    query: (col) => col.filtered("_id in $0", resultIds),
    }, [resultIds]);
```